### PR TITLE
feat: uikit-v4-components

### PR DESCRIPTION
**Pull request by NaingAmity**

### CODE REVIEW

File: src/v4/component/StoryKit/index.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import Story from './src/Story';
+
+export * from './src/interfaces';
+
+export const InstaStory = Story;
+
+export default InstaStory;
```
Comment (frankent) [inline src/v4/component/StoryKit/index.ts:7]: Why `InstaStory` ?

File: src/v4/component/MyStories/index.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+      ) : globalStoriesItems?.length > 0 ? (
+        <InstaStory
+          data={globalStoriesItems}
+          duration={15}
+          onStorySeen={({ story }) => story.markAsSeen()}
+        />
+      ) : null}
+    </View>
+  );
+}
```
Comment (frankent) [inline src/v4/component/MyStories/index.tsx:123]:
```suggestion
  if (loading) {
    return (
      <View style={styles.container}>
        <ContentLoader
          height={70}
          speed={1}
          width={100}
          backgroundColor={'#d2d2d2'}
          foregroundColor={'#eee'}
          viewBox="-10 7 100 30"
        >
          <Circle cx="25" cy="25" r="25" />
        </ContentLoader>
      </View>
    );
  }
  
  if (globalStoriesItems?.length > 0 ) {
    return (
      <View style={styles.container}>
        <InstaStory
          data={globalStoriesItems}
          duration={15}
          onStorySeen={({ story }) => story.markAsSeen()}
        />
      </View>
    );
  }
  
  return null
}
```

File: src/v4/component/StoryKit/src/Story.tsx [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+        } else {
+          setCurrentPage(newPage);
+          cube?.current?.scrollTo(newPage);
+        }
+      }
+    }
+  }
+
+  const renderStoryList = () =>
+    selectedData.map((x, i) => {
```
Comment (frankent) [inline src/v4/component/StoryKit/src/Story.tsx:115]: Can we make x and i more readable ?

File: src/v4/component/StoryKit/src/StoryListItem.tsx [hunk 4]
```diff
@@ -0,0 +1,495 @@
...
+  useEffect(() => {
+    let isPrevious = !!prevCurrentPage && prevCurrentPage > currentPage;
+    if (isPrevious) {
+      setCurrent(content.length - 1);
+    } else {
+      setCurrent(0);
+    }
+
+    let data = [...content];
+    data.map((x, i) => {
```
Comment (frankent) [inline src/v4/component/StoryKit/src/StoryListItem.tsx:126]: this section too `x` and `i`

File: src/v4/screen/CommunityHome/styles.ts [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+import type { MyMD3Theme } from 'src/providers/amity-ui-kit-provider';
+
+export const useStyles = () => {
+  const theme = useTheme() as MyMD3Theme;
+  const styles = StyleSheet.create({
+    container: {
+      flex: 1,
+      backgroundColor: theme.colors.background,
+      // alignItems: 'center',
+      // justifyContent: 'center',
```
Comment (frankent) [inline src/v4/screen/CommunityHome/styles.ts:12]: Please dont forget to remove this comment too

File: src/v4/screen/Feed/index.tsx [hunk 6]
```diff
@@ -0,0 +1,162 @@
+import React, {
+  forwardRef,
+  useCallback,
+  useImperativeHandle,
+  useState,
+} from 'react';
+
+// import { useTranslation } from 'react-i18next';
```
Comment (frankent) [inline src/v4/screen/Feed/index.tsx:8]:
```suggestion
```

File: src/v4/screen/GlobalFeed/index.tsx [hunk 7]
```diff
@@ -0,0 +1,108 @@
+import React, { useCallback, useRef, useState } from 'react';
+
+// import { useTranslation } from 'react-i18next';
```
Comment (frankent) [inline src/v4/screen/GlobalFeed/index.tsx:3]:
```suggestion
```

why we dont need this anymore ?

File: src/v4/screen/Home/index.tsx [hunk 8]
```diff
@@ -0,0 +1,99 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import * as React from 'react';
+import { useEffect, useState } from 'react';
+// import { useTranslation } from 'react-i18next';
```
Comment (frankent) [inline src/v4/screen/Home/index.tsx:4]:
```suggestion
```

File: .gitignore [hunk 9]
```diff
@@ -44,6 +44,7 @@ android.iml
 # Cocoapods
 #
 example/ios/Pods
+example/Gemfile.lock
```
Comment (bonanaaaaaa) [inline .gitignore:47]: remove this file

**Review by frankent — CHANGES_REQUESTED** [review]
Please check all my comments and questions

File: src/v4/component/CommunityStories/index.tsx [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+    }, [communityId, getStories])
+  );
+
+  const [communityStories, setCommunityStories] = useState([]);
+  const formatStory = useCallback(() => {
+    const isSeen = stories.every((story) => story.isSeen);
+    const storyData = stories?.map((item: TAmityStory) => {
+      return {
+        story_id: item.storyId,
+        story_image: `https://api.${apiRegion}.amity.co/api/v3/files/${item?.data?.fileId}/download?size=full`,
```
Comment (bonanaaaaaa) [inline src/v4/component/CommunityStories/index.tsx:51]: use sdk instead

File: src/v4/component/MyStories/index.tsx [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+        const avatarFileId = community.avatarFileId;
+        const displayName = community.displayName;
+        const isSeen = items.every((item) => item.isSeen);
+
+        const storyData = items.map((item) => {
+          return {
+            story_id: item.storyId,
+            story_image: item?.imageData?.fileUrl,
+            swipeText: '',
+            onPress: () => console.log('story 1 swiped'),
```
Comment (bonanaaaaaa) [inline src/v4/component/MyStories/index.tsx:72]: should we remove `console.log`

File: src/v4/component/StoryKit/src/Story.tsx
Comment (bonanaaaaaa) [inline src/v4/component/StoryKit/src/Story.tsx:1]: should we use `npm` package instead

File: src/v4/component/MyStories/index.tsx [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+  isPublic: boolean;
+  isOfficial: boolean;
+  hasStories: boolean;
+}
+
+export default function MyStories() {
+  const styles = useStyles();
+  const { apiRegion } = useAuth();
+  const [globalStoriesItems, setGlobalStoriesItems] = useState<
+    IUserStory<Record<string, any>>[]
```
Comment (bonanaaaaaa) [inline src/v4/component/MyStories/index.tsx:22]: replace `any` type with a more proper type

File: src/v4/hook/useStory.ts
Comment (bonanaaaaaa) [inline src/v4/hook/useStory.ts:1]: separate into multiple hooks

**Pull request merged**